datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -2766,6 +2766,72 @@ where
     )?))
 }
 
+/// array_reverse SQL function
+pub fn array_reverse(arg: &[ArrayRef]) -> Result<ArrayRef> {
+    if arg.len() != 1 {
+        return exec_err!("array_reverse needs one argument");
+    }
+
+    match &arg[0].data_type() {
+        DataType::List(field) => {
+            let array = as_list_array(&arg[0])?;
+            general_array_reverse::<i32>(array, &field)
+        }
+        DataType::LargeList(field) => {
+            let array = as_large_list_array(&arg[0])?;
+            general_array_reverse::<i64>(array, &field)
+        }
+        DataType::Null => Ok(arg[0].clone()),
```

Comment:
We do not handle `FixedSizeList` here, which will be solved by type coercion.

Reply:
Ah makes sense. So this would be done in a later PR, if I understand correctly?

Else if it should already work with `FixedSizeList` then can add that as a test case in slt

Reply:
It is still working on other pr, so I added a TODO test in slt.